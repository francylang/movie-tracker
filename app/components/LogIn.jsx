import React, { Component } from 'react';
import { Redirect } from 'react-router';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange(event, type) {
    let value = event.target.value;
    if (type === 'email') {
      value = value.toLowerCase();
    }
    this.setState({
      [type]: value
    });
  }


  render() {
    if (this.props.userStatus === true) {
      return <Redirect to='/'/>;
    }

    return (
      <div>
        <form>
          <h2>Welcome to Movie Tracker</h2>
          <h2>LOGIN</h2>
          <input
            type='text'
            placeholder='Email'
            onChange={ (event) => this.handleChange(event, 'email' ) }
          />
          <input
            type='password'
            placeholder='Password'
            onChange={ (event) => this.handleChange(event, 'password' ) }
          />
          <input
            className='form-button'
            type='submit'
            onClick={
              (event) => {
                event.preventDefault();
                //pass state to action that makes fetch post request to get user back
                const userObj = Object.assign({}, this.state);
                this.props.retrieveUser(userObj);
              }
            }
          />
        </form>
      </div>
    );
  }
}

export default Login;
