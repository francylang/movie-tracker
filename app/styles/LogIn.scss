.login {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 100;
  transform: translate(-50%, -50%);
  background-color: white;
  height: 340px;
  width: 400px;
  border-radius: 5px;
  padding: 30px;
  margin: 40px 0 0;
  box-shadow: 5px 10px 10px #bdbdbd;
}

.login h2 {
  text-align: center;
  color: #D23321;

}

.login input {
  margin-top: 30px;
}

.password-input,
.email-input,
.name-input {
  width: 100%;
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #bdbdbd;
  font-size: 20px;
  font-weight: lighter;
  margin: 20px 0;
}
.password-input:focus,
.email-input:focus,
.name-input:focus {
  outline: none;
  border-bottom: 2px solid #D23321;
}

.form-button {
  background-color: #D23321;
  border: none;
  border-radius: 10px;
  color: #fff;
  padding: 8px 20px;
  font-size: 16px;
  margin: 20px 0;
  width: 100%;
}

.form-button:focus {
  outline: none;
}

.form-button:hover {
  background-color: #aa2f1e;
}

.form-button:disabled {
  background-color: #bdbdbd;
}

.heading {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.error {
  margin-top: 10px;
  color: darkgray;
}
