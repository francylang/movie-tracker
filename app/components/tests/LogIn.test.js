import React from "react";
import LogIn from '../LogIn';
import { shallow } from "enzyme";

describe('Login page', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<LogIn userStatus=''
      retrieveUser={mockFn}/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have an empty default state', () => {
    expect(wrapper.state().email).toEqual('');
    expect(wrapper.state().password).toEqual('');
  });

  it('should render three inputs', () => {
    const inputs = wrapper.find('input');

    expect(inputs.length).toEqual(3);
  });

  it('should change state on input', () => {
    const emailInput = wrapper.find('.email-input');
    const passwordInput = wrapper.find('.password-input');

    emailInput.simulate('change', { target: { value: 'lolabrenner@gmail.com' } });
    passwordInput.simulate('change', { target: { value: 'complete' } });
    expect(wrapper.state().email).toEqual('lolabrenner@gmail.com');
    expect(wrapper.state().password).toEqual('complete');
  });

  it('should run function on submit', () => {
    const button = wrapper.find('input').last();

    button.simulate('click', { preventDefault() {} });
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an error if passed one', () => {
    const altWrapper = shallow(<LogIn userStatus=''
      retrieveUser={mockFn}
      loginError={true}/>);
    const error = altWrapper.find('h4');

    expect(error.text()).toEqual('Email and Password do not match');
    expect(altWrapper).toMatchSnapshot();
  });
});
