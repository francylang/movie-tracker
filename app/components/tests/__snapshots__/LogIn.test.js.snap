// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    retrieveUser={[Function]}
    userStatus=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <h2>
          LOGIN
        </h2>
        <h4
          className="error"
        />
        <input
          className="email-input"
          onChange={[Function]}
          placeholder="Email"
          type="text"
        />
        <input
          className="password-input"
          onChange={[Function]}
          placeholder="Password"
          type="text"
        />
        <input
          className="form-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        />
      </form>,
      "className": "login",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            LOGIN
          </h2>,
          <h4
            className="error"
          />,
          Array [
            <input
              className="email-input"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />,
            <input
              className="password-input"
              onChange={[Function]}
              placeholder="Password"
              type="text"
            />,
          ],
          <input
            className="form-button"
            disabled={true}
            onClick={[Function]}
            type="submit"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "LOGIN",
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "error",
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": "email",
          "nodeType": "host",
          "props": Object {
            "className": "email-input",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": "password",
          "nodeType": "host",
          "props": Object {
            "className": "password-input",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "form-button",
            "disabled": true,
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <h2>
            LOGIN
          </h2>
          <h4
            className="error"
          />
          <input
            className="email-input"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <input
            className="password-input"
            onChange={[Function]}
            placeholder="Password"
            type="text"
          />
          <input
            className="form-button"
            disabled={true}
            onClick={[Function]}
            type="submit"
          />
        </form>,
        "className": "login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              LOGIN
            </h2>,
            <h4
              className="error"
            />,
            Array [
              <input
                className="email-input"
                onChange={[Function]}
                placeholder="Email"
                type="text"
              />,
              <input
                className="password-input"
                onChange={[Function]}
                placeholder="Password"
                type="text"
              />,
            ],
            <input
              className="form-button"
              disabled={true}
              onClick={[Function]}
              type="submit"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "LOGIN",
            },
            "ref": null,
            "rendered": "LOGIN",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "error",
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": "email",
            "nodeType": "host",
            "props": Object {
              "className": "email-input",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": "password",
            "nodeType": "host",
            "props": Object {
              "className": "password-input",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "form-button",
              "disabled": true,
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Login page should render an error if passed one 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    loginError={true}
    retrieveUser={[Function]}
    userStatus=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <h2>
          LOGIN
        </h2>
        <h4
          className="error"
        >
          Email and Password do not match
        </h4>
        <input
          className="email-input"
          onChange={[Function]}
          placeholder="Email"
          type="text"
        />
        <input
          className="password-input"
          onChange={[Function]}
          placeholder="Password"
          type="text"
        />
        <input
          className="form-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        />
      </form>,
      "className": "login",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            LOGIN
          </h2>,
          <h4
            className="error"
          >
            Email and Password do not match
          </h4>,
          Array [
            <input
              className="email-input"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />,
            <input
              className="password-input"
              onChange={[Function]}
              placeholder="Password"
              type="text"
            />,
          ],
          <input
            className="form-button"
            disabled={true}
            onClick={[Function]}
            type="submit"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "LOGIN",
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Email and Password do not match",
            "className": "error",
          },
          "ref": null,
          "rendered": "Email and Password do not match",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": "email",
          "nodeType": "host",
          "props": Object {
            "className": "email-input",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": "password",
          "nodeType": "host",
          "props": Object {
            "className": "password-input",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "form-button",
            "disabled": true,
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <h2>
            LOGIN
          </h2>
          <h4
            className="error"
          >
            Email and Password do not match
          </h4>
          <input
            className="email-input"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <input
            className="password-input"
            onChange={[Function]}
            placeholder="Password"
            type="text"
          />
          <input
            className="form-button"
            disabled={true}
            onClick={[Function]}
            type="submit"
          />
        </form>,
        "className": "login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              LOGIN
            </h2>,
            <h4
              className="error"
            >
              Email and Password do not match
            </h4>,
            Array [
              <input
                className="email-input"
                onChange={[Function]}
                placeholder="Email"
                type="text"
              />,
              <input
                className="password-input"
                onChange={[Function]}
                placeholder="Password"
                type="text"
              />,
            ],
            <input
              className="form-button"
              disabled={true}
              onClick={[Function]}
              type="submit"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "LOGIN",
            },
            "ref": null,
            "rendered": "LOGIN",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Email and Password do not match",
              "className": "error",
            },
            "ref": null,
            "rendered": "Email and Password do not match",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": "email",
            "nodeType": "host",
            "props": Object {
              "className": "email-input",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": "password",
            "nodeType": "host",
            "props": Object {
              "className": "password-input",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "form-button",
              "disabled": true,
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
